extensions/guacamole-vault/modules/guacamole-vault-ksm/src/main/java/org/apache/guacamole/vault/ksm/secret/KsmClient.java

```diff
@@ -250,13 +291,38 @@ private void validateCache() throws GuacamoleException {
                 String hostname = recordService.getHostname(record);
                 addRecordForHost(record, hostname);
 
+                // ... and domain
+                String domain = recordService.getDomain(record);
+                addRecordForDomain(record, domain);
+
+                // Fetch the username
+                String username = recordService.getUsername(record);
+
+                // If domains should be split out from usernames
+                if (username != null && confService.getSplitWindowsUsernames()) {
+
+                    // Attempt to split the domain of the username
+                    WindowsUsername usernameAndDomain = (
+                            WindowsUsername.splitWindowsUsernameFromDomain(username));
```

Comment:
A couple points:

* Should this be abstracted away such that the caller need not concern themselves with whether they need to manually split domain from username?
* As currently written, `username` will be one of the following:
  * The raw username containing the domain, which will not match the split username and domain that would be part of the connection parameters.
  * The split username (with the domain stored in a custom field of the record), which will ambiguously match any user with that username regardless of domain.

I think we need to somehow take this into account and ensure that indexed usernames include the domain, and it would be good if the caller need not manually know whether they need to handle the splitting.

Reply:
What caller are you referring to, specifically?

Reply:
This extension.

Reply:
Ah, yes good call. The functionality was a bit goofed, but it should be better now. The `KEEPER_USER_` tokens will now ALSO match on domain (if configured to do so. false by default). 

The behavior seems a lot more sane now - let me know what you think.

Reply:
This definitely looks much better.